core/discovery/reducer/in.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reducer
+
+import (
+	"github.com/mysteriumnetwork/node/market"
+)
+
+// In matches proposal if field value exists in array
```

Comment:
Such documentation makes it hard to understand what these functions do. Since most of them does nothing, just returns other function, which does actual job. I would suggest fixing docs as follows: 
```suggestion
// In returns a matcher for checking if proposal's field value exists in given array of values
```